include/dlaf/eigensolver/bt_band_to_tridiag/mc.h

```diff
@@ -135,8 +138,103 @@ auto updateE(pika::threads::thread_priority priority, WSender&& tile_w, W2Sender
       pika::execution::experimental::start_detached();
 }
 
+struct Helper {
+  Helper(const SizeType b, const SizeType nrefls, matrix::Distribution dist,
+         const GlobalElementIndex offset)
+      : nrefls_(nrefls), input_spec_{dist.tileElementIndex(offset),
+                                     {std::min(b, dist.size().rows() - offset.row()),
+                                      std::min(b, dist.size().cols() - offset.col())}} {
+    const TileElementIndex& sub_offset = input_spec_.origin;
+
+    const SizeType tile_row = dist.globalTileFromGlobalElement<Coord::Row>(offset.row());
+    const bool isLastRow = tile_row == dist.nrTiles().rows() - 1;
+
+    const SizeType max_size = dist.size().rows() - offset.row() - 1;
+    if (isLastRow && max_size < 2 * b - 1) {
+      mode_ = Mode::SINGLE_PART;
+      parts_[0] = part_t{max_size, nrefls, sub_offset.row() + 1};
+    }
+    else {
+      const SizeType mb = dist.blockSize().rows();
+      if (mb - sub_offset.row() - 1 >= 2 * b - 1) {
+        mode_ = Mode::SINGLE_FULL;
+        parts_[0] = part_t{2 * b - 1, nrefls, sub_offset.row() + 1};
+      }
+      else {
+        mode_ = Mode::DOUBLE_FULL;
+        const SizeType mb2 = dist.size().rows() - offset.row() - b;
+        parts_[0] = part_t{b - 1, nrefls, sub_offset.row() + 1};
+        parts_[1] = part_t{std::min(b, mb2), nrefls, 0, b - 1};
+      }
+    }
+
+    input_spec_ = {sub_offset,
```

Comment:
Why do you set it again with the same values?

Reply:
Good catch, leftover of the refactoring 😄